### ISSUE TITLE
[sge:lockHost] Error locking host

### ISSUE THREAD

**Issue by porcaro33** [aws/aws-parallelcluster#1073]
**Environment:**
 - aws-parallelcluster-2.3.1
 - OS: CentOS7
 - Scheduler: SGE
 - Master instance type: m5.2xlarge
 - Compute instance type: c5.18xlarge

**Bug description and how to reproduce:**
When I submit mpi jobs, ParallelCluster will launch the computing nodes, but the hosts are never registered into SGE. SGE keep no-host status. So the submitted jobs is kept in qw status forever.

**Additional context:**
 - post-install script will install basic tool gcc, traceroute, wget ...etc
 - the computing node are configured as scale down in 15min. But the compute node termination was started in 10min according to nodewatcher log.
 - I can see the error messge "[sge:lockHost] Error locking host...". But I am not sure the reason.

nodewatcher log:
2019-05-24 21:44:48,033 INFO [nodewatcher:_poll_instance_status] Instance had no job for the past 10 minute(s)
2019-05-24 21:44:48,086 ERROR [utils:_run_command] Command '['/opt/sge/bin/lx-amd64/qstat', '-g', 'd', '-s', 'p', '-u', '*']' returned non-zero exit status 1
2019-05-24 21:44:48,087 INFO [nodewatcher:_poll_instance_status] Encountered an error while polling queue for pending jobs. Not terminating instance
denied: host "ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal" is not an admin host
2019-05-24 21:44:48,162 ERROR [utils:_run_command] Command '['/opt/sge/bin/lx-amd64/qmod', '-d', 'all.q@ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal']' returned non-zero exit status 1
2019-05-24 21:44:48,162 ERROR [sge:lockHost] Error locking host ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal
2019-05-24 21:45:03,235 ERROR [utils:_run_command] Command '['/opt/sge/bin/lx-amd64/qstat', '-g', 't', '-l', 'hostname=ip-XXX-XXX-XXX-XXX.us-west-2.compute.internal', '-u', '*']' returned non-zero exit status 1
2019-05-24 21:45:03,357 INFO [nodewatcher:_maintain_size] DesiredCapacity is 7, MinSize is 0
2019-05-24 21:45:03,357 INFO [nodewatcher:_self_terminate] Self terminating i-XXXXXXXXXXXX

**Comment by demartinofra** [comment]
Hi,

Thank you for reporting this. In order to debug the root cause we need additional details:
- configuration file without any credentials or personal data.
- please attach the following files `/var/log/nodewatcher` from the Compute node and `/var/log/jobwatcher` and `/var/log/sqswatcher` from the Master node
- and `/var/log/cfn-init.log` from compute node

It seems like the sqswatcher never attached this node to the scheduler and this might be due to the post-install script taking longer than the configured idletime. However we need the above mentioned files to confirm this.

As a side note this behaviour will be largely improved in the upcoming version of ParallelCluster

Francesco

**Comment by porcaro33** [comment]
Hi Francesco,

I attached those logfiles and config file as following

[(node)cfn-init.log](https://github.com/aws/aws-parallelcluster/files/3226268/node.cfn-init.log)
[nodewatcher2.txt](https://github.com/aws/aws-parallelcluster/files/3226269/nodewatcher2.txt)
[config.txt](https://github.com/aws/aws-parallelcluster/files/3226278/config.txt)
[jobwatcher.txt](https://github.com/aws/aws-parallelcluster/files/3226279/jobwatcher.txt)
[sqswatcher.txt](https://github.com/aws/aws-parallelcluster/files/3226280/sqswatcher.txt)

**Comment by demartinofra** [comment]
Thanks for providing the additional info.

By looking at the sqswatcher log file I can see nodes are terminated (`EC2_INSTANCE_TERMINATE` event) before they are even able to join the cluster. This might be due to the post_install script taking more time than the allowed scaledown_idletime. From your config I can see you are trying to set this timeout to 15 minutes but you forgot to add a `scaling_settings = custom` section under cluster config, which makes it default to 10 minutes. 
This behaviour will be largely improved with the upcoming version of ParallelCluster since the idletime timer will be started only after the post_install script has been completed. For the moment an increase of the scaledown_idletime should be a viable workaround.

Moreover rather than using long post_install scripts which slow down the scaling of your cluster I would recommend you to look into the `pcluster createami` command: https://aws-parallelcluster.readthedocs.io/en/latest/tutorials/02_ami_customization.html#build-a-custom-aws-parallelcluster-ami

**Comment by porcaro33** [comment]
oh, I missed `scaling_settings = custom` in cluster section on config file.
So `scaledown_idletime = 15` did not worked on this cluster.
Let me try again with correct config.

**Comment by demartinofra** [comment]
Once you have retried please let us know if that fixed your issue. Thanks!

**Comment by porcaro33** [comment]
The issue was fixed by extending scaledown_idletime. Thank you!
